lib/cog/models/ecto_json.ex

```diff
@@ -89,5 +131,25 @@ defmodule Cog.Models.EctoJson do
     non_struct
   end
 
+  defp render_field(:permissions = field, struct, render_policy, acc) do
+    perms = Enum.group_by(do_render(Map.get(struct, field), render_policy), &(Map.get(&1, :namespace)))
+    Map.put(acc, field, Enum.into(Map.keys(perms), %{}, &scrub(&1, perms, :namespace)))
+  end
+  defp render_field(field, struct, render_policy, acc),
+    do: Map.put(acc, field, do_render(Map.get(struct, field), render_policy))
+
+  defp scrub(index, list_maps, key) when is_map(list_maps) do
+    case is_list(list_maps[index]) do
+      true ->
+        {index, Enum.map(list_maps[index], fn (data_map) ->
+          {_value, scrubbed} = Map.pop(data_map, key)
+          scrubbed
```

Comment:
Since your dropping the `value`, `Map.delete` might be a tiny bit nicer here. You might be able to just send `&Map.delete(&1, key)` into the `Enum.map` actually.